File: src/main/java/com/msg/gauth/domain/enums/UserRole.java
package com.msg.gauth.domain.enums;

import org.springframework.security.core.GrantedAuthority;

public enum UserRole implements GrantedAuthority {
    ROLE_STUDENT, ROLE_TEACHER, ROLE_ADMIN;

    @Override
    public String getAuthority() {
        return name();
    }
}

File: src/main/kotlin/com/msg/gauth/domain/user/enums/UserRole.kt
package com.msg.gauth.domain.user.enums

import org.springframework.security.core.GrantedAuthority

enum class UserRole : GrantedAuthority {
    ROLE_STUDENT, ROLE_TEACHER, ROLE_ADMIN, ROLE_GRADUATE;

    override fun getAuthority(): String =
        name
}
